PWG/EMCAL/EMCALtasks/AliEmcalMCTrackSelector.cxx

```diff
@@ -351,11 +356,49 @@ AliEmcalMCTrackSelector* AliEmcalMCTrackSelector::AddTaskMCTrackSelector(TString
   }
 
   // Check the analysis type using the event handlers connected to the analysis manager.
-  if (!mgr->GetInputEventHandler()) {
+  //==============================================================================
+  AliVEventHandler* handler = mgr->GetInputEventHandler();
+  if (!handler)
+  {
     ::Error("AddTaskMCTrackSelector", "This task requires an input event handler");
     return NULL;
   }
 
+  enum EDataType_t {
+    kUnknown,
+    kESD,
+    kAOD,
+    kMCgen
+  };
+
+  TString trackName(nTrackCont);
+
+  bool isContSpecified = false;
+  if(trackName == "usedefault") isContSpecified = false;
+  else isContSpecified = true;
+
+  EDataType_t dataType = kUnknown;
+  if (handler->InheritsFrom("AliESDInputHandler")) {
+    dataType = kESD;
+  }
+  else if (handler->InheritsFrom("AliAODInputHandler")) {
+    dataType = kAOD;
+  } else if (handler->InheritsFrom("AliMCGenHandler")) {
+    dataType = kMCgen;
+  }
+
+  if (trackName == "usedefault") {
+    if (dataType == kESD) {
+      trackName = "Tracks";
+    }
+    else if (dataType == kAOD) {
+      trackName = "tracks";
+    }
```

Comment:
The tasks works purely on MC particles. "Tracks" and "tracks" are for ESD/AOD tracks.